widget-src/components/log/LogEditing.tsx

```diff
@@ -10,10 +10,74 @@ import { LinkForm } from './LinkForm';
 import { AddLink } from './AddLink';
 import { ActionCloseIcon } from '../../svgs/ActionCloseIcon';
 import { displayDate } from '../../utilities/Utils';
+import { LinkType } from '../../types/LinkTypes';
 
-const { widget } = figma;
+const { widget, saveVersionHistoryAsync } = figma;
 const { AutoLayout, Text } = widget;
 
+async function saveToVersionHistory(
+  name: string,
+  type: string,
+  description: string,
+  links: LinkType[] | undefined,
+  showName: boolean,
+  showTypes: boolean,
+  showVersion: boolean,
+  version?: string,
+) {
+  const formattedLinks = links?.map(link => `${link.label}\n${link.url}`).join('\n\n') || '';
+  if (!description.trim() && !formattedLinks.trim()) {
+    return;
+  }
+
+  let typeFormatted = '';
+  if (showTypes) {
+    switch (type) {
+      case 'none':
+        break;
+      case 'added': // catch legacy logs with "added" as default type
+        typeFormatted = '';
+        break;
```

Comment:
You don't need to set this to an empty string, it already is one, you can stack the 'none' and 'added' cases together and just break.

Also, I wish we didn't need to repeat this logic since we are using something super similar in the Type component, but not sure if that is worth the refactor at this moment.